app/components/oauth/oauth.controller.ts

```diff
@@ -56,15 +62,31 @@ export class OAuthController implements angular.IController {
               // persist the user
               _this.AuthService.setUser(user, token, ttl);
 
-              // redirect to original destination (or default to './')
-              var destination = then || './';
-              if (URI(destination).is('absolute')) {
-                _this.authLogger.log('OAuthController, invalid absolute redirect', destination);
-                destination = './';
+              // Check if this is the first time this user has logged in, if so launch the tour
+              var loggedInKey: string = "hasLoggedIn/" + user.metadata.name;
+              var hasLoggedIn: boolean = localStorage.getItem(loggedInKey) === 'true';
+
+              if (hasLoggedIn) {
+                // redirect to original destination (or default to './')
+                var destination = then || './';
+                if (URI(destination).is('absolute')) {
+                  _this.authLogger.log('OAuthController, invalid absolute redirect', destination);
+                  destination = './';
+                }
+                _this.authLogger.log('OAuthController, redirecting', destination);
+                _this.$location.replace();
+                _this.$location.url(destination);
+              } else {
+                localStorage.setItem(loggedInKey, 'true');
+                _this.$location.replace();
+                _this.$location.url('./');
+                this.$timeout(function() {
+                  let tourConfig = _this.Constants.LANDING_PAGE_TOUR_STEPS;
+                  if (tourConfig && tourConfig.steps) {
+                    _this.GuidedTourService.startTour(tourConfig);
```

Comment:
What if we:

- Allow shutting of the auto-tour via the config
- Don't add the menu item or the button if the constant does not exist
- Auto start the tour when the landing page is navigated to and the user has not yet seen it.

As for the upgrade, this was discussed with @beanh66 during development and the idea is that this is new content and it is valid to offer a tour of it.